Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs

```diff
@@ -37,82 +38,50 @@ public static class DecalRegistry {
-                    speed = float.Parse(attrs["speed"].Value);
-                float amplitude = 1f;
-                if (attrs["amplitude"] != null)
-                    amplitude = float.Parse(attrs["amplitude"].Value);
-                int sliceSize = 1;
-                if (attrs["sliceSize"] != null)
-                    sliceSize = int.Parse(attrs["sliceSize"].Value);
-                float sliceSinIncrement = 1f;
-                if (attrs["sliceSinIncrement"] != null)
-                    sliceSinIncrement = float.Parse(attrs["sliceSinIncrement"].Value);
+                float offset = attrs["offset"] != null ? float.Parse(attrs["offset"].Value) : 0f;
+                float speed = attrs["speed"] != null ? float.Parse(attrs["speed"].Value) : 1f;
+                float amplitude = attrs["amplitude"] != null ? float.Parse(attrs["amplitude"].Value) : 1f;
+                int sliceSize = attrs["sliceSize"] != null ? int.Parse(attrs["sliceSize"].Value) : 1;
+                float sliceSinIncrement = attrs["sliceSinIncrement"] != null ? float.Parse(attrs["sliceSinIncrement"].Value) : 1f;
                 bool easeDown = attrs["easeDown"] != null ? bool.Parse(attrs["easeDown"].Value) : false;
                 bool onlyIfWindy = attrs["onlyIfWindy"] != null ? bool.Parse(attrs["onlyIfWindy"].Value): false;
                 ((patch_Decal)decal).MakeBanner(speed, amplitude, sliceSize, sliceSinIncrement, easeDown, offset, onlyIfWindy);
             }},
             { "solid", delegate(Decal decal, XmlAttributeCollection attrs) {
-                float x = 0;
-                if (attrs["x"] != null)
-                    x = float.Parse(attrs["x"].Value);
-                float y = 0;
-                if (attrs["y"] != null)
-                    y = float.Parse(attrs["y"].Value);
-                float width = 16;
-                if (attrs["width"] != null)
-                    width = float.Parse(attrs["width"].Value);
-                float height = 16;
-                if (attrs["height"] != null)
-                    height = float.Parse(attrs["height"].Value);
-                int index = SurfaceIndex.ResortRoof;
-                if (attrs["index"] != null)
-                    index = int.Parse(attrs["index"].Value);
+                float x = attrs["x"] != null ? float.Parse(attrs["x"].Value) : 0f;
+                float y = attrs["y"] != null ? float.Parse(attrs["y"].Value) : 0f;
+                float width = attrs["width"] != null ? float.Parse(attrs["width"].Value) : 16f;
+                float height = attrs["height"] != null ? float.Parse(attrs["height"].Value) : 16f;
+                int index = attrs["index"] != null ? int.Parse(attrs["index"].Value) : SurfaceIndex.ResortRoof;
                 bool blockWaterfalls = attrs["blockWaterfalls"] != null ? bool.Parse(attrs["blockWaterfalls"].Value) : true;
                 ((patch_Decal)decal).MakeSolid(x, y, width, height, index, blockWaterfalls);
```

Comment:
currently `MakeSolid` creates a solid with `safe: true`; is this something that might benefit from configurability (for example if a decal is a solid staticmover?)

Reply:
I don't think anyone's requested this but probably wouldn't hurt to add. Maybe in a different PR since we discussed other changes to the tag anyway

Reply:
might as well put it in my decal registry parity pr

Reply:
now in https://github.com/EverestAPI/Everest/pull/463